src/vm/node_modules/VM.js

```diff
@@ -8688,11 +8697,11 @@ function installZone(payload, log, callback)
             // Create any filesystems now that have the 'create' flag set.
             // Note that currently if you reprovision, all these created volumes
             // will be destroyed and recreated as well since they're children of
-            // the zoneroot. This only currently works with "docker" VMs.
+            // the zoneroot. This does not work with HVMs.
             if (receiving
                 || reprovisioning
                 || !payload.hasOwnProperty('add_filesystems')
-                || !vmobj.docker) {
+                || vmobj.hvm) {
```

Comment:
Per MM, test on LX.  If it works, great.  If it doesn't, add a `|| vmobj.lx` or whatever-it-is check.

Reply:
Notes added to jira.